lib/openstudio/extension.rb

```diff
@@ -103,142 +102,142 @@ def self.all_extensions
       end
       return result.uniq
     end
-    
+
     ##
     # Module method to return measure directories from all extensions
     ##
-    #  @return [Array]  Array of measure directories
+    #  @return [Array]: Array of measure directories
     def self.all_measure_dirs
       result = []
-      self.all_extensions.each do |obj|
+      all_extensions.each do |obj|
         begin
           dir = obj.new.measures_dir
           result << dir if dir
-        rescue
+        rescue StandardError
         end
       end
       return result.uniq
     end
-    
+
     ##
     # Module method to return measure resource directories from all extensions
     ##
     #  @return [Array]  Array of measure resource directories
-    def self.all_measure_resource_dirs
+    def self.all_core_dirs
```

Comment:
Previously we were assuming that each derived gem could have it's own measure_resource dir and those files would be copied to measures using them.  Do we assume the same here with a core dir?  Or is the extension gem the only one with a core dir?

Previously files in extension-gem/measure_resources would be copied into measures with resource files having the same name.  For example, extension-gem/lib/measure_resources/os_lib_geometry.rb would be copied onto model-articulation-gem/lib/measures/some_measure/resources/os_lib_geometry.rb.  Do we still want that to happen?

Reply:
Should the core lib insert itself into measures recursively as needed?

Reply:
I think there is a single core dir in openstudio_extension gems.  That directory does not exist in other gems. 

However, extension-gem/measure_resources would be copied into measures with resource files having the same name (for now anyway, to preserve measures that can be used directly / pushed to BCL).

Reply:
i think we should update the measures now to include the extension core library methods instead of copying the files into the measures. Will that work with the CLI?

Reply:
That will work with the CLI but will not work with older versions of the CLI, non-bundle workflows (e.g. ComStock/Restock), or with the App/PAT/Server.  So we would have to maintain two copies of the measures, one in openstudio-measures to support all those other use cases.

Reply:
If we did clean up now we could also fix the module names of the core libs.  However, having two copies of the measures is going to make everyone very uneasy.